expression/builtin_json.go

```diff
@@ -140,6 +112,14 @@ func jsonModify(args []types.Datum, mt json.ModifyType, sc *variable.StatementCo
 			vs = append(vs, args[i])
 		}
 	}
+	if args[0].Kind() == types.KindNull || argsAnyNull(pes) {
+		return d, nil
+	}
+
+	j, err := datum2JSON(args[0], sc)
```

Comment:
no, there are to logic: `ToMysqlJSON` and `convertToMysqlJSON`. they are differents.